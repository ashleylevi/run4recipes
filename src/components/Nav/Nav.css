.link {

  /* display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: 0 auto;
  margin-top: 40px; */
}

.nav-bar {
  display: flex;
  border-top: 4px solid #3D323E;
  border-bottom: 4px solid #3D323E;
  height: 50px;
  width: 1050px;
  margin: 0 auto;
  margin-top: 40px;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  margin: 0 auto;
  margin-top: 40px;
}

.search-div {
  display: flex;
  flex-direction: flex-end;
  margin-left: 197px;
}

.nav-link {
  border: none;
  font-family: Raleway;
  font-weight: bold;
  font-size: 15px;
  color: #3D323E;
  margin-right: 80px;
}

.nav-link:hover {
  cursor: pointer;
  color: #D5AD47;
}

.active {
  font-weight: 700;
}