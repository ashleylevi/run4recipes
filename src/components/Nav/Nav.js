import React from 'react';
import { Link } from 'react-router-dom';
import './Nav.css';

const Nav = () => {
  return(
    <div className="nav-bar">
      <Link to={'/allRecipes'}>
            <button className='all-btn'>All Recipes</button>
      </Link>
      <Link to={'/pasta'}>
            <button className='pasta-btn'>Pasta</button>
      </Link>
      <Link to={'/potato'}>
            <button className='potatoes-btn'>Potatoes</button>
      </Link>
      <Link to={'/bread'}>
            <button className='bread-btn'>Bread</button>
      </Link>
      <Link to={'/favorites'}>
            <button className='favorites-btn'>Favorites</button>
      </Link>
    </div>
  )
}
  
export default Nav;
