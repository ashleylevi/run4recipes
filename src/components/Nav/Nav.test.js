import React from 'react';
import { shallow } from 'enzyme';
import Nav from './Nav';

describe('Nav', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Nav />);
  });

  it('Should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});