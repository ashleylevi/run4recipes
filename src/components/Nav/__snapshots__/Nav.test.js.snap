// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Nav />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="link"
        >
          <NavLink
            activeClassName="active"
            activeStyle={
              Object {
                "color": "#D5AD47",
              }
            }
            aria-current="page"
            className="nav-link"
            to="/allRecipes"
          >
            All Recipes
          </NavLink>
          <NavLink
            activeClassName="active"
            activeStyle={
              Object {
                "color": "#D5AD47",
              }
            }
            aria-current="page"
            className="nav-link"
            to="/pasta"
          >
            Pasta
          </NavLink>
          <NavLink
            activeClassName="active"
            activeStyle={
              Object {
                "color": "#D5AD47",
              }
            }
            aria-current="page"
            className="nav-link"
            to="/potato"
          >
            Potatoes
          </NavLink>
          <NavLink
            activeClassName="active"
            activeStyle={
              Object {
                "color": "#D5AD47",
              }
            }
            aria-current="page"
            className="nav-link"
            to="/bread"
          >
            Bread
          </NavLink>
          <NavLink
            activeClassName="active"
            activeStyle={
              Object {
                "color": "#D5AD47",
              }
            }
            aria-current="page"
            className="nav-link"
            to="/favorites"
          >
            Favorites
          </NavLink>
        </div>,
        <div
          className="search"
        >
          <Connect(Search) />
        </div>,
      ],
      "className": "nav-bar",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/allRecipes"
            >
              All Recipes
            </NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/pasta"
            >
              Pasta
            </NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/potato"
            >
              Potatoes
            </NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/bread"
            >
              Bread
            </NavLink>,
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/favorites"
            >
              Favorites
            </NavLink>,
          ],
          "className": "link",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#D5AD47",
              },
              "aria-current": "page",
              "children": "All Recipes",
              "className": "nav-link",
              "to": "/allRecipes",
            },
            "ref": null,
            "rendered": "All Recipes",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#D5AD47",
              },
              "aria-current": "page",
              "children": "Pasta",
              "className": "nav-link",
              "to": "/pasta",
            },
            "ref": null,
            "rendered": "Pasta",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#D5AD47",
              },
              "aria-current": "page",
              "children": "Potatoes",
              "className": "nav-link",
              "to": "/potato",
            },
            "ref": null,
            "rendered": "Potatoes",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#D5AD47",
              },
              "aria-current": "page",
              "children": "Bread",
              "className": "nav-link",
              "to": "/bread",
            },
            "ref": null,
            "rendered": "Bread",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "activeStyle": Object {
                "color": "#D5AD47",
              },
              "aria-current": "page",
              "children": "Favorites",
              "className": "nav-link",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Connect(Search) />,
          "className": "search",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="link"
          >
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/allRecipes"
            >
              All Recipes
            </NavLink>
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/pasta"
            >
              Pasta
            </NavLink>
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/potato"
            >
              Potatoes
            </NavLink>
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/bread"
            >
              Bread
            </NavLink>
            <NavLink
              activeClassName="active"
              activeStyle={
                Object {
                  "color": "#D5AD47",
                }
              }
              aria-current="page"
              className="nav-link"
              to="/favorites"
            >
              Favorites
            </NavLink>
          </div>,
          <div
            className="search"
          >
            <Connect(Search) />
          </div>,
        ],
        "className": "nav-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#D5AD47",
                  }
                }
                aria-current="page"
                className="nav-link"
                to="/allRecipes"
              >
                All Recipes
              </NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#D5AD47",
                  }
                }
                aria-current="page"
                className="nav-link"
                to="/pasta"
              >
                Pasta
              </NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#D5AD47",
                  }
                }
                aria-current="page"
                className="nav-link"
                to="/potato"
              >
                Potatoes
              </NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#D5AD47",
                  }
                }
                aria-current="page"
                className="nav-link"
                to="/bread"
              >
                Bread
              </NavLink>,
              <NavLink
                activeClassName="active"
                activeStyle={
                  Object {
                    "color": "#D5AD47",
                  }
                }
                aria-current="page"
                className="nav-link"
                to="/favorites"
              >
                Favorites
              </NavLink>,
            ],
            "className": "link",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#D5AD47",
                },
                "aria-current": "page",
                "children": "All Recipes",
                "className": "nav-link",
                "to": "/allRecipes",
              },
              "ref": null,
              "rendered": "All Recipes",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#D5AD47",
                },
                "aria-current": "page",
                "children": "Pasta",
                "className": "nav-link",
                "to": "/pasta",
              },
              "ref": null,
              "rendered": "Pasta",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#D5AD47",
                },
                "aria-current": "page",
                "children": "Potatoes",
                "className": "nav-link",
                "to": "/potato",
              },
              "ref": null,
              "rendered": "Potatoes",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#D5AD47",
                },
                "aria-current": "page",
                "children": "Bread",
                "className": "nav-link",
                "to": "/bread",
              },
              "ref": null,
              "rendered": "Bread",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "activeStyle": Object {
                  "color": "#D5AD47",
                },
                "aria-current": "page",
                "children": "Favorites",
                "className": "nav-link",
                "to": "/favorites",
              },
              "ref": null,
              "rendered": "Favorites",
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Connect(Search) />,
            "className": "search",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
